packages/framework/data-object-base/src/lazyLoadedDataObjectFactory.ts

```diff
@@ -52,26 +52,35 @@ export class LazyLoadedDataObjectFactory<T extends LazyLoadedDataObject> impleme
         existing: boolean,
     ): Promise<FluidDataStoreRuntime> {
         const runtimeClass = mixinRequestHandler(
-            async (request: IRequest) => {
-                const router = await instance;
+            async (request: IRequest, rt: FluidDataStoreRuntime) => {
+                // This factory knows that the entrypoint on the data stores it creates is a LazyLoadedDataObject
+                // (or child of it) because it passed it in (see the call to new runtimeClass(...) below), so it
+                // can cast safely here.
+                const maybeIFluidLoadable: FluidObject<IProvideFluidLoadable> = rt;
+                const router = (await maybeIFluidLoadable.IFluidLoadable?.handle?.get() as LazyLoadedDataObject);
```

Comment:
i think we could make the casting a bit better here

Reply:
`const maybeRouter: FluidObject<IFluidRouter> & IFluidLoadable | undefined`

Reply:
i know its verbose, but its how our typing current works. as we talked about we can make this better

Reply:
I see it'll work, I went with the current approach because all the code paths ended up calling `this.ctor()` and thus returning a `T extends LazyLoadedDataObject`, so it seemed safe to do it within a class that can put guarantees on the object it's creating. Would you still rather use the more generic approach?

Reply:
i really hate as casting, as it can easily become a crutch that seeps out into other places, so if we can avoid it, i prefer we do.